PencilCase/PencilCase/scripts/CustomBindings/ListviewBinding.ts

```diff
@@ -1,11 +1,25 @@
 ﻿class ListviewBinding implements KnockoutBindingHandler {
-    public update(element: any, valueAccessor: () => any, allowBindingAccessor: KnockoutAllBindingsAccessor, viewModel: any, bindingContext: KnockoutBindingContext) {
+    public init(element: any, valueAccessor: () => any, allowBindingAccessor: KnockoutAllBindingsAccessor, viewModel: any, bindingContext: KnockoutBindingContext) {
+        /*
+        Implemented listview custom binding handler.
+        Due to dynamically inserted(KO binding) DOMs are not enhanced by JQM,
+        so need this handler here, once the product collection changed, re-enhance the DOM.
+
+        Note: Note that I did not add any export keyword in this class and the Binding class in Binding.ts.
+        In this way, it makes the members in this file are globally available. Just like the JQuery, KO libraries.
+        */
+
         var value = valueAccessor();
+        let subscription: KnockoutSubscription = null;
         if (ko.isObservable(value)) {
-            value.subscribe((newValue: any)=>{
+            subscription = value.subscribe((newValue: any)=>{
                 $(element).listview("refresh");
             });
         }
+
+        ko.utils.domNodeDisposal.addDisposeCallback(element, () => {
```

Comment:
Need to dispose the subscription when DOM dispose, or it will cause unexpected behavior and memory leak.